doc/record.md

```diff
@@ -0,0 +1,100 @@
+# Record format
+Records are intended to make exchange of RDF safer and easier. The design has arised when working with facility design data, but we expect it to be useful for many cases. Specifically, use cases with high-velocity append-only data is probably not a good match. 
+
+Two existing, safe, approaches to exchanging RDF is to either exchange lists of individual triples to deleted and inserted, or to always exchange the full graph. Both these approaches have valid use cases, but have major disadvantages in those we work with. We claim that records are useful for any situation that is not handled well by these two extremes.
```

Comment:
Two existing, tried approaches to exchanging RDF are full graph exchange and triple assertion exchange: a (change)set of triples each either inserted or deleted. 
- Full graph exchange is semantically portable (no context is assumed, the exchanged graph can be utilized by anyone at any time, not only the intended recipient at the time of exchange). 
- Full graph exchange is inflexible. In order to communicate a small change to a large graph, the whole graph must be transmitted. For a given size of graph and rate of change this can prove cost-prohibitive.
- Triple assertion exchange is flexible and prudent. Changesets only contain the actual changes (however small) to the graph (however large). Changesets are trivially combined and decomposed.
- Triple assertion exchange is largely non-portable. A changeset is only meaningfull in the context of the _intended_ graph to which it applies, or together with an ordered series of the previous changesets.

We present an ontology of the RDF Record as a middle ground between full graph- and triple assertion exchange, providing portability of exchanged sub-graphs (Records) combined with the (somewhat limited) flexibility and prudence of communicating change by transmitting (and thus "replacing") only the changed sub-graphs.

Our main design goals are:
- Ease of integration with various domain specific information systems at their native level of composition for domain objects (1 updated domain object -> 1 record)
- Immutable history (a Record is never deleted, only ever superseeded by a new Record)
- Transparency of provenance (a Record's history and "chain of custory" can be traced back to the origin)
- Branching and merging of record histories
- Portability of partial graphs (i.e. a Record can be useful in contexts where it's history or sub-/superrecords are neither known nor needed).

Reply:
Har lagt det inn i motivation.md